security/subsystem/src/main/java/org/jboss/as/security/SecurityDomainResourceDefinition.java

```diff
@@ -226,19 +227,20 @@ protected void executeRuntimeStep(OperationContext context, ModelNode operation)
      * @throws OperationFailedException if the service is not available, or the thread was interrupted.
      */
     private static void waitForService(final ServiceController<?> controller) throws OperationFailedException {
-        if (controller.getState() == ServiceController.State.UP) return;
-
-        final StabilityMonitor monitor = new StabilityMonitor();
-        monitor.addController(controller);
         try {
-            monitor.awaitStability(100, MILLISECONDS);
+            final CountDownLatch latch = new CountDownLatch(1);
+            final LifecycleListener listener = new LifecycleListener() {
+                @Override
+                public void handleEvent(ServiceController<?> controller, LifecycleEvent event) {
+                    latch.countDown();
+                }
+            };
+            controller.addListener(listener);
+            latch.await();
         } catch (final InterruptedException e) {
             Thread.currentThread().interrupt();
             throw SecurityLogger.ROOT_LOGGER.interruptedWaitingForSecurityDomain(controller.getName().getSimpleName());
-        } finally {
-            monitor.removeController(controller);
-        }
-
+        } catch (final Throwable ignored) {}
```

Comment:
Shouldn't the lifecycle listener be removed in a finally block?

Reply:
Of course it must be removed @pferraro - thanks for spotting that out.